<h1>Problem Statement

<h3>Health Profile by Age and Gender: Examine the health profiles of individuals by age and gender. Analyze various health parameters such as blood pressure, cholesterol levels, fasting blood sugar, and body mass index (BMI) across different age groups and genders. Identify any age or gender-related variations in health indicators and explore potential implications for health insurance and healthcare services.

In [ ]:
#importing required libraries & modules

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
%matplotlib inline
import matplotlib.pyplot as plt

<b><h1>Introduction</h1></b>
 <br>
Analysis of Health checkup data of people collected for south Korea National Health Insurance.<br>
In Korea, everyone is compulsorily required to join the National Health Insurance. The National Health Insurance Service (NHIS), which manages national health insurance, provides basic health checkups to subscribers every year.

<b>Collected dataset from kaggel-</b><br>
<b>link</b> - https://www.kaggle.com/datasets/hongseoi/health-checkup-result?select=2020.csv

In [3]:
#importing dataset
health_data = pd.read_csv('C:\Mayur Dusane\Data science study\Sample Datasets for projects\Health insurance.csv')

In [4]:
pd.set_option('display.max_columns', 500)
health_data.head()

YEAR  IDV_ID  AREA_CODE  SEX  AGE_GROUP  HEIGHT  WEIGHT  WAIST  SIGHT_LEFT  \
0  2019       1         11    1         13     165      65   90.0         1.5   
1  2019       2         48    2          5     155      45   64.0         1.2   
2  2019       3         41    1         13     160      70   87.0         0.9   
3  2019       4         28    1         12     170      60   85.0         0.4   
4  2019       5         41    1         10     170      70   81.2         0.7   

   SIGHT_RIGHT  HEAR_LEFT  HEAR_RIGHT  BP_HIGH  BP_LWST   BLDS  TOT_CHOLE  \
0          1.2        1.0         1.0    112.0     70.0   93.0        NaN   
1          0.5        1.0         1.0    114.0     76.0   89.0        NaN   
2          0.8        1.0         1.0    109.0     73.0  100.0      232.0   
3          1.2        1.0         1.0    131.0     85.0   85.0        NaN   
4          0.7        1.0         1.0    112.0     66.0   91.0        NaN   

   TRIGLYCERIDE  HDL_CHOLE  LDL_CHOLE   HMG  OLIG_PROTE_CD  CREATININE  \
0           NaN        NaN        NaN  15.5            1.0         1.0   
1           NaN        NaN        NaN  14.0            1.0         0.5   
2         114.0       57.0      152.0  15.6            1.0         1.0   
3           NaN        NaN        NaN  14.5            1.0         0.7   
4           NaN        NaN        NaN  14.9            1.0         0.9   

   SGOT_AST  SGPT_ALT  GAMMA_GTP  SMK_STAT  DRK_YN  HCHK_CE_IN  CRS_YN  \
0      18.0      32.0       47.0         2       0           1     0.0   
1      15.0       8.0       13.0         1       1           0     NaN   
2      30.0      21.0       17.0         1       1           0     NaN   
3      17.0      19.0       32.0         2       1           0     NaN   
4      19.0      24.0       19.0         2       0           0     NaN   

   TTR_YN        DATE  
0     1.0  31-12-2019  
1     NaN  31-12-2019  
2     NaN  31-12-2019  
3     NaN  31-12-2019  
4     NaN  31-12-2019

# Description of Columns


| feature name | description | form of expression |
| :-: | -: | :-: |
| YEAR 	| Base year of the information 	| YYYY |
| IDV_ID	| Serial number assigned to subscriber 	| N |
| AREA_CODE	| Residency code of examinee	| N |
| SEX	| Gender	| 1: male, 2:female |
| AGE_GROUP	| A code that categorizes the examinee's age into 5-year-olds based on the year.	| 1 = 5 Years |
| HEIGHT	| Examiner's height (in units of 5 cm)	| cm |
| WEIGHT	| Examiner's weight (in units of 5 kg)	| Kg |
| WAIST	| examiner's waist circumference	| cm |
| SIGHT_LEFT	| Eyesight of the examinee's left eye	| (0.1~2.5, eyesight < 0.1 == 0.1, blind==9.9) |
| SIGHT_RIGHT	| Eyesight of the examinee's right eye	| (0.1~2.5, eyesight < 0.1 == 0.1, blind==9.9) |
| BP_HIGH	| The examiner's systolic blood pressure	| mmHg |
| BP_LWST	| Diastolic blood pressure of examinee	| mmHg |
| BLDS	| Pre-meal blood glucose of the examinee. The concentration of glucose per 100 ml of blood	| mg/dL |
| TOT_CHOLE	| Sum of ester and non-ester cholesterol in serum. Normal values are 150 to 250 mg/dL	| mg/dL |
| TRIGLYCERIDE	| Amount of simple lipids or neutral lipids. Normal values are 30 to 135 mg/dL	| mg/dL |
| HDL_CHOLE	| The amount of cholesterol contained in HDL. Normal values are 30 to 65 mg/dL.	| mg/dL |
| LDL_CHOLE	| The amount of cholesterol contained in LDL. If it is 170 mg/dL or higher, hyper-LDLemia is diagnosed.	| mg/dL |
| CREATININE	| Serum concentration of creatinine, the dehydration of creatine. Increases and decreases in creatinine are not related to food, but to muscle development and exercise. Normal values are 0.8 to 1.7 mg/dL.	| mg/dL |
| HMG	| It is a pigment-protein present in blood and blood cells, composed of globin and heme, and plays a role as an oxygen carrier in the blood.	| g/dL |
| OLIG_PROTE_CD	| excretion of protein in the urine	| 1(-), 2(±), 3(+1), 4(+2), 5(+3), 6(+4) |
| SGOT_AST	| Levels on blood tests that indicate liver function. Concentrations increase when liver cells, heart, kidney, brain, and muscle cells are damaged. Normal value is 0~40IU/L	| IU/L |
| SGPT_ALT	| Levels in blood tests that indicate liver function. ALT mainly exists only in hepatocytes, and its concentration increases when hepatocytes are damaged. Normal values are 0 to 40 IU/L	| IU/L |
| GAMMA_GTP	| 	| IU/L |
| SMK_STAT | Whether or not the examinee's smoking status	| 1: No Smoking,      2: Smoking |
| DRK_YN | Whether or not the examinee's drinking status	| 0: No Drinking,      1: Drinking |
| HCHK_CE_IN | Whether or not the examinee chose oral examination.	| 0: Not Tested,      1: Tested |
| CRS_YN | Whether or not the examinee has dental caries	| 0:None, 1:present |
| TTH_MSS_YN | Existence of missing teeth of the examinee	| 0:None, 1:present |
| ODT_TRB_YN | Whether or not the examinee has dental abrasion	| 0:None, 1:present |
| WSDM_DIS_YN | Whether or not the examinee has wisdom teeth	| 0:None, 1:present |
| TTR_YN | Whether or not the examinee has calculus	| 0:None, 1:present |
| DATE | Date the data was created	| YYYY-MM-DD |

In [5]:
health_data.columns

Index(['YEAR', 'IDV_ID', 'AREA_CODE', 'SEX', 'AGE_GROUP', 'HEIGHT', 'WEIGHT',
       'WAIST', 'SIGHT_LEFT', 'SIGHT_RIGHT', 'HEAR_LEFT', 'HEAR_RIGHT',
       'BP_HIGH', 'BP_LWST', 'BLDS', 'TOT_CHOLE', 'TRIGLYCERIDE', 'HDL_CHOLE',
       'LDL_CHOLE', 'HMG', 'OLIG_PROTE_CD', 'CREATININE', 'SGOT_AST',
       'SGPT_ALT', 'GAMMA_GTP', 'SMK_STAT', 'DRK_YN', 'HCHK_CE_IN', 'CRS_YN',
       'TTR_YN', 'DATE'],
      dtype='object')

<h1>Data Cleaning and Preprocessing

In [6]:
#Renaming column names for better understanding
Clean_Health_Data = health_data.rename(columns={'IDV_ID':'ID', 'HEIGHT':'HEIGHT(CM)', 'WEIGHT':'WEIGHT(KG)','WAIST':'WAIST(CM)',
                            'SIGHT_LEFT':'EYE-SIGHT(LEFT)','SIGHT_RIGHT':'EYE-SIGHT(RIGHT)','HEAR_LEFT':'HEARING(LEFT)',
                            'HEAR_RIGHT':'HEARING(RIGHT)','BP_HIGH':'SYSTOLIC BP','BP_LWST':'DIASTOLIC BP','BLDS':'FASTING BLOOD SUGAR',
                            'TOT_CHOLE':'TOTAL CHOLESTEROL','HDL_CHOLE':'HDL CHOLESTEROL','LDL_CHOLE':'LDL CHOLESTEROL',
                            'HMG':'HEMOGLOBIN','OLIG_PROTE_CD':'URINE PROTEIN','SGOT_AST':'AST','SGPT_ALT':'ALT','GAMMA_GTP':'GAMMA GTP',
                            'SMK_STAT':'SMOKING','DRK_YN':'ALCOHOL','HCHK_CE_IN':'ORAL EXAMINATION','CRS_YN':'DENTAL CARIES','TTR_YN':'CALCULUS TEETH'})

In [7]:
Clean_Health_Data.head(2)

YEAR  ID  AREA_CODE  SEX  AGE_GROUP  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)  \
0  2019   1         11    1         13         165          65       90.0   
1  2019   2         48    2          5         155          45       64.0   

   EYE-SIGHT(LEFT)  EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  \
0              1.5               1.2            1.0             1.0   
1              1.2               0.5            1.0             1.0   

   SYSTOLIC BP  DIASTOLIC BP  FASTING BLOOD SUGAR  TOTAL CHOLESTEROL  \
0        112.0          70.0                 93.0                NaN   
1        114.0          76.0                 89.0                NaN   

   TRIGLYCERIDE  HDL CHOLESTEROL  LDL CHOLESTEROL  HEMOGLOBIN  URINE PROTEIN  \
0           NaN              NaN              NaN        15.5            1.0   
1           NaN              NaN              NaN        14.0            1.0   

   CREATININE   AST   ALT  GAMMA GTP  SMOKING  ALCOHOL  ORAL EXAMINATION  \
0         1.0  18.0  32.0       47.0        2        0                 1   
1         0.5  15.0   8.0       13.0        1        1                 0   

   DENTAL CARIES  CALCULUS TEETH        DATE  
0            0.0             1.0  31-12-2019  
1            NaN             NaN  31-12-2019

<b>Dropping Unwanted Columns.<br>
Removing columns ('YEAR','AREA_CODE','ORAL EXAMINATION', 'DENTAL CARIES','CALCULUS TEETH','DATE').

In [8]:
#Removing unwanted Columns
Clean_Health_Data.drop(['YEAR','AREA_CODE','ORAL EXAMINATION', 'DENTAL CARIES','CALCULUS TEETH','DATE'], axis=1, inplace=True)

In [9]:
Clean_Health_Data.head(2)

ID  SEX  AGE_GROUP  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)  EYE-SIGHT(LEFT)  \
0   1    1         13         165          65       90.0              1.5   
1   2    2          5         155          45       64.0              1.2   

   EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  SYSTOLIC BP  DIASTOLIC BP  \
0               1.2            1.0             1.0        112.0          70.0   
1               0.5            1.0             1.0        114.0          76.0   

   FASTING BLOOD SUGAR  TOTAL CHOLESTEROL  TRIGLYCERIDE  HDL CHOLESTEROL  \
0                 93.0                NaN           NaN              NaN   
1                 89.0                NaN           NaN              NaN   

   LDL CHOLESTEROL  HEMOGLOBIN  URINE PROTEIN  CREATININE   AST   ALT  \
0              NaN        15.5            1.0         1.0  18.0  32.0   
1              NaN        14.0            1.0         0.5  15.0   8.0   

   GAMMA GTP  SMOKING  ALCOHOL  
0       47.0        2        0  
1       13.0        1        1

<b><h1>Finding Columns with Missing Values(Null Values).</h1></b><br>
<b>In below cell, Columns with missing values are showing True value as a output and columns with no missing value are showing False.

In [10]:
#Checking missing values
Clean_Health_Data.isnull().any()

ID                     False
SEX                    False
AGE_GROUP              False
HEIGHT(CM)             False
WEIGHT(KG)             False
WAIST(CM)               True
EYE-SIGHT(LEFT)         True
EYE-SIGHT(RIGHT)        True
HEARING(LEFT)           True
HEARING(RIGHT)          True
SYSTOLIC BP             True
DIASTOLIC BP            True
FASTING BLOOD SUGAR     True
TOTAL CHOLESTEROL       True
TRIGLYCERIDE            True
HDL CHOLESTEROL         True
LDL CHOLESTEROL         True
HEMOGLOBIN              True
URINE PROTEIN           True
CREATININE              True
AST                     True
ALT                     True
GAMMA GTP               True
SMOKING                False
ALCOHOL                False
dtype: bool

<h1>Checking for Duplicate Records in Dataframe.</h1><br>
<b>Used 'ID' column to find duplicate records in dataframe. 'ID' Column has unique values.<br>
Output is showing False which means Dataframe doesn't have duplicate records.

In [11]:
#Checking duplicate values
True in Clean_Health_Data['ID'].duplicated().tolist()

False

In [12]:
#Removing ID CLoumn as it is no longer needed
Clean_Health_Data.drop('ID',axis=1,inplace=True)

In [13]:
#Removing missing values
Clean_Health_Data.dropna(axis=0, inplace=True)
#dropped all rows containing null values

In [15]:
#Validating null values
Clean_Health_Data.isnull().any().any()

False

In [16]:
Clean_Health_Data.head()

SEX  AGE_GROUP  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)  EYE-SIGHT(LEFT)  \
2     1         13         160          70       87.0              0.9   
5     1          8         170         110      107.8              1.2   
6     1          6         175          70       81.3              0.9   
10    1         11         170          75       86.5              0.7   
12    2         11         150          65       94.0              1.2   

    EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  SYSTOLIC BP  \
2                0.8            1.0             1.0        109.0   
5                1.5            1.0             1.0        139.0   
6                1.0            1.0             1.0        120.0   
10               0.2            1.0             1.0        140.0   
12               1.0            1.0             1.0        123.0   

    DIASTOLIC BP  FASTING BLOOD SUGAR  TOTAL CHOLESTEROL  TRIGLYCERIDE  \
2           73.0                100.0              232.0         114.0   
5           87.0                108.0              215.0         142.0   
6           72.0                 90.0              148.0          51.0   
10          80.0                 87.0              193.0         144.0   
12          69.0                 82.0              252.0         163.0   

    HDL CHOLESTEROL  LDL CHOLESTEROL  HEMOGLOBIN  URINE PROTEIN  CREATININE  \
2              57.0            152.0        15.6            1.0         1.0   
5              37.0            150.0        14.8            1.0         0.9   
6              45.0             93.0        13.1            1.0         0.9   
10             42.0            122.0        13.9            1.0         1.1   
12             43.0            176.0        13.9            1.0         0.8   

     AST   ALT  GAMMA GTP  SMOKING  ALCOHOL  
2   30.0  21.0       17.0        1        1  
5   42.0  44.0      129.0        1        1  
6   14.0  11.0       30.0        2        1  
10  21.0  22.0       37.0        2        0  
12  22.0  13.0       39.0        1        1

<h1><b>Checking and Changing Datatypes of Columns</b></h1><br>

In [17]:
#Checking data types of all columns
Clean_Health_Data.dtypes

SEX                      int64
AGE_GROUP                int64
HEIGHT(CM)               int64
WEIGHT(KG)               int64
WAIST(CM)              float64
EYE-SIGHT(LEFT)        float64
EYE-SIGHT(RIGHT)       float64
HEARING(LEFT)          float64
HEARING(RIGHT)         float64
SYSTOLIC BP            float64
DIASTOLIC BP           float64
FASTING BLOOD SUGAR    float64
TOTAL CHOLESTEROL      float64
TRIGLYCERIDE           float64
HDL CHOLESTEROL        float64
LDL CHOLESTEROL        float64
HEMOGLOBIN             float64
URINE PROTEIN          float64
CREATININE             float64
AST                    float64
ALT                    float64
GAMMA GTP              float64
SMOKING                  int64
ALCOHOL                  int64
dtype: object

In [18]:
#Changing datatypes of columns
Clean_Health_Data[['HEARING(LEFT)','HEARING(RIGHT)','SYSTOLIC BP','DIASTOLIC BP','FASTING BLOOD SUGAR','TOTAL CHOLESTEROL',
                   'TRIGLYCERIDE','HDL CHOLESTEROL','LDL CHOLESTEROL', 'URINE PROTEIN', 'CREATININE', 'AST', 'ALT',
                   'GAMMA GTP', 'SMOKING', 'ALCOHOL']] = Clean_Health_Data[['HEARING(LEFT)','HEARING(RIGHT)','SYSTOLIC BP', 'DIASTOLIC BP', 'FASTING BLOOD SUGAR','TOTAL CHOLESTEROL', 
                   'TRIGLYCERIDE', 'HDL CHOLESTEROL', 'LDL CHOLESTEROL', 'URINE PROTEIN','CREATININE', 'AST', 'ALT', 
                   'GAMMA GTP', 'SMOKING', 'ALCOHOL']].astype(int)

In [19]:
#Validate the data
Clean_Health_Data.dtypes
Clean_Health_Data.head()

SEX  AGE_GROUP  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)  EYE-SIGHT(LEFT)  \
2     1         13         160          70       87.0              0.9   
5     1          8         170         110      107.8              1.2   
6     1          6         175          70       81.3              0.9   
10    1         11         170          75       86.5              0.7   
12    2         11         150          65       94.0              1.2   

    EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  SYSTOLIC BP  \
2                0.8              1               1          109   
5                1.5              1               1          139   
6                1.0              1               1          120   
10               0.2              1               1          140   
12               1.0              1               1          123   

    DIASTOLIC BP  FASTING BLOOD SUGAR  TOTAL CHOLESTEROL  TRIGLYCERIDE  \
2             73                  100                232           114   
5             87                  108                215           142   
6             72                   90                148            51   
10            80                   87                193           144   
12            69                   82                252           163   

    HDL CHOLESTEROL  LDL CHOLESTEROL  HEMOGLOBIN  URINE PROTEIN  CREATININE  \
2                57              152        15.6              1           1   
5                37              150        14.8              1           0   
6                45               93        13.1              1           0   
10               42              122        13.9              1           1   
12               43              176        13.9              1           0   

    AST  ALT  GAMMA GTP  SMOKING  ALCOHOL  
2    30   21         17        1        1  
5    42   44        129        1        1  
6    14   11         30        2        1  
10   21   22         37        2        0  
12   22   13         39        1        1

<b>Creating 'AGE' column with the help of 'AGE_GROUP' column.</b>

In [20]:
#Replacing AGE_GROUP column with AGE column
#AGE=AGE_GROUP * 5
#AGE will be multiple of 5 according to dataset

In [21]:
Clean_Health_Data['AGE_GROUP']=Clean_Health_Data['AGE_GROUP']*5
Clean_Health_Data.rename(columns={'AGE_GROUP':'AGE'}, inplace=True)

In [22]:
Clean_Health_Data.head(2)

SEX  AGE  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)  EYE-SIGHT(LEFT)  \
2    1   65         160          70       87.0              0.9   
5    1   40         170         110      107.8              1.2   

   EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  SYSTOLIC BP  DIASTOLIC BP  \
2               0.8              1               1          109            73   
5               1.5              1               1          139            87   

   FASTING BLOOD SUGAR  TOTAL CHOLESTEROL  TRIGLYCERIDE  HDL CHOLESTEROL  \
2                  100                232           114               57   
5                  108                215           142               37   

   LDL CHOLESTEROL  HEMOGLOBIN  URINE PROTEIN  CREATININE  AST  ALT  \
2              152        15.6              1           1   30   21   
5              150        14.8              1           0   42   44   

   GAMMA GTP  SMOKING  ALCOHOL  
2         17        1        1  
5        129        1        1

<b>Replacing SEX column numerical values with categorical values.</b>

In [23]:
#Replacing SEX column numerical values with categorical values
Clean_Health_Data['SEX'] = Clean_Health_Data['SEX'].replace({1:'M',2:'F'})

<b>Changing 'SMOKING' column values.</b><br>

In [24]:
#Changing SMOKING columns values
Clean_Health_Data['SMOKING'] = Clean_Health_Data['SMOKING'].replace({1:0,2:1})

<b>Changing 'HEARING(LEFT)' and 'HEARING(RIGHT)' columns values to 0 (No hearing impairment) and 1 (Hearing impairment).</b>

In [25]:
#Changing HEARING(LEFT) values to 0 (No hearing impairment) and 1 (Hearing impairment)
Clean_Health_Data['HEARING(LEFT)'] = Clean_Health_Data['HEARING(LEFT)'].replace({1:0,2:1})

In [26]:
#Changing HEARING(RIGHT) values to 0 (No hearing impairment) and 1 (Hearing impairment)
Clean_Health_Data['HEARING(RIGHT)'] = Clean_Health_Data['HEARING(RIGHT)'].replace({1:0,2:1})

<b>Creating 'BMI' Column using 'HEIGHT(CM)' and 'WEIGHT(KG)' columns.<b/>

In [27]:
#Creating BMI Column using HEIGHT(CM) and WEIGHt(KG) columns
Clean_Health_Data['BMI']= Clean_Health_Data['WEIGHT(KG)']/(Clean_Health_Data['HEIGHT(CM)']/100)**2

In [28]:
#Rounding off BMI Column values upto one decimal
Clean_Health_Data['BMI'] = Clean_Health_Data['BMI'].round(decimals=1)

<b>Creating 'W/H RATIO' (Waist to height ratio) column using 'WAIST(CM)' and 'HEIGHT(CM)' columns.<b>

In [29]:
#Creating WAIST TO HEIGHT RATIO
Clean_Health_Data['W/H RATIO'] = Clean_Health_Data['WAIST(CM)']/Clean_Health_Data['HEIGHT(CM)']
Clean_Health_Data['W/H RATIO'] = Clean_Health_Data['W/H RATIO'].round(decimals=2)

In [30]:
Clean_Health_Data.head()

SEX  AGE  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)  EYE-SIGHT(LEFT)  \
2    M   65         160          70       87.0              0.9   
5    M   40         170         110      107.8              1.2   
6    M   30         175          70       81.3              0.9   
10   M   55         170          75       86.5              0.7   
12   F   55         150          65       94.0              1.2   

    EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  SYSTOLIC BP  \
2                0.8              0               0          109   
5                1.5              0               0          139   
6                1.0              0               0          120   
10               0.2              0               0          140   
12               1.0              0               0          123   

    DIASTOLIC BP  FASTING BLOOD SUGAR  TOTAL CHOLESTEROL  TRIGLYCERIDE  \
2             73                  100                232           114   
5             87                  108                215           142   
6             72                   90                148            51   
10            80                   87                193           144   
12            69                   82                252           163   

    HDL CHOLESTEROL  LDL CHOLESTEROL  HEMOGLOBIN  URINE PROTEIN  CREATININE  \
2                57              152        15.6              1           1   
5                37              150        14.8              1           0   
6                45               93        13.1              1           0   
10               42              122        13.9              1           1   
12               43              176        13.9              1           0   

    AST  ALT  GAMMA GTP  SMOKING  ALCOHOL   BMI  W/H RATIO  
2    30   21         17        0        1  27.3       0.54  
5    42   44        129        0        1  38.1       0.63  
6    14   11         30        1        1  22.9       0.46  
10   21   22         37        1        0  26.0       0.51  
12   22   13         39        0        1  28.9       0.63

<b>Creating HEARING IMPAIRMENT column by comparing HEARING(LEFT) and HEARING(RIGHT).

In [31]:
#Creating HEARING IMPAIRMENT column by comparing HEARING(LEFT) and HEARING(RIGHT)
Clean_Health_Data['HEARING IMPAIRMENT'] = Clean_Health_Data.apply(lambda x: 1 if (x['HEARING(LEFT)']==1 or x['HEARING(RIGHT)']==1) else 0, axis=1)

<b><h1>Rearranging Columns Positions

In [32]:
Clean_Health_Data.columns

Index(['SEX', 'AGE', 'HEIGHT(CM)', 'WEIGHT(KG)', 'WAIST(CM)',
       'EYE-SIGHT(LEFT)', 'EYE-SIGHT(RIGHT)', 'HEARING(LEFT)',
       'HEARING(RIGHT)', 'SYSTOLIC BP', 'DIASTOLIC BP', 'FASTING BLOOD SUGAR',
       'TOTAL CHOLESTEROL', 'TRIGLYCERIDE', 'HDL CHOLESTEROL',
       'LDL CHOLESTEROL', 'HEMOGLOBIN', 'URINE PROTEIN', 'CREATININE', 'AST',
       'ALT', 'GAMMA GTP', 'SMOKING', 'ALCOHOL', 'BMI', 'W/H RATIO',
       'HEARING IMPAIRMENT'],
      dtype='object')

In [33]:
#Rearranging columns
Final_Data = Clean_Health_Data.loc[:,['SEX', 'AGE', 'HEIGHT(CM)', 'WEIGHT(KG)', 'WAIST(CM)', 'BMI','W/H RATIO',
       'EYE-SIGHT(LEFT)', 'EYE-SIGHT(RIGHT)', 'HEARING(LEFT)',
       'HEARING(RIGHT)','HEARING IMPAIRMENT', 'SYSTOLIC BP', 'DIASTOLIC BP', 'FASTING BLOOD SUGAR',
       'TOTAL CHOLESTEROL', 'TRIGLYCERIDE', 'HDL CHOLESTEROL',
       'LDL CHOLESTEROL', 'HEMOGLOBIN', 'URINE PROTEIN', 'CREATININE', 'AST',
       'ALT', 'GAMMA GTP', 'SMOKING', 'ALCOHOL',
       ]]

In [34]:
Final_Data.head(2)

SEX  AGE  HEIGHT(CM)  WEIGHT(KG)  WAIST(CM)   BMI  W/H RATIO  \
2   M   65         160          70       87.0  27.3       0.54   
5   M   40         170         110      107.8  38.1       0.63   

   EYE-SIGHT(LEFT)  EYE-SIGHT(RIGHT)  HEARING(LEFT)  HEARING(RIGHT)  \
2              0.9               0.8              0               0   
5              1.2               1.5              0               0   

   HEARING IMPAIRMENT  SYSTOLIC BP  DIASTOLIC BP  FASTING BLOOD SUGAR  \
2                   0          109            73                  100   
5                   0          139            87                  108   

   TOTAL CHOLESTEROL  TRIGLYCERIDE  HDL CHOLESTEROL  LDL CHOLESTEROL  \
2                232           114               57              152   
5                215           142               37              150   

   HEMOGLOBIN  URINE PROTEIN  CREATININE  AST  ALT  GAMMA GTP  SMOKING  \
2        15.6              1           1   30   21         17        0   
5        14.8              1           0   42   44        129        0   

   ALCOHOL  
2        1  
5        1

<b><h1>Exporting Dataframe to CSV file

In [36]:
#Saving cleaned dataframe into csv file
Final_Data.to_csv('C:\Mayur Dusane\Data science study\Sample Datasets for projects\Final_Data.csv',index=False)